docs/testing/unit-tests.md

```diff
@@ -1,3 +1,160 @@
 # Unit Tests
 
-TBD
+As well fostering code quality, well-written unit tests provide a fine way for developers to quickly understand how to run a piece of code, and what it does under various conditions. It's important therefore to maintain high standards not just in our source code, but in our tests as well. 
+
+This guide is intended as a quick reference of common tools and conventions we use for Calypso unit testing. You'll be able to glean a great deal from browsing through Calypso's existing tests, but we'll try to keep this page up to date with the most recent standards.
+
+## Table of contents
+* [Running tests](running-tests)
+* [Writing tests](writing-tests)
+    * [Describing tests](#describing-tests)
+    * [Folder structure](#folder-structure)
+    * [Mocking HTTP requests with nock](#mocking-http-requests-with nock)
+
+## What should I unit test?
+We assume you've Googled the 'why' of unit testing, but sometimes deciding __what__ to unit test is more difficult. 
+
+If your code were to break or receive unexpected input, what side-effects would the rest of the application experience?
+
+## Running tests
+
+To execute all client side tests, run the following command in the root of the project folder:
+
+```bash
+npm run test-client
+```
+
+You can also run all tests in a specific directory or matching pattern, for example:
+
+```bash
+# run test suites for all files in a specific folder
+npm run test-client client/lib/domains
+
+# run test suites for all files matching pattern
+npm run test-client client/*/domains
+```
+
+To run a __specific suite or test__, append `.only() ` to the `describe` and `test` functions:
+
+```javascript
+describe.only( 'just run this suite', function() {
+	test( 'should run these tests', function() {
+		// your test
+	} );
+
+	test.only( 'should only run this one test', function() {
+		// just run this test if the only is present
+	} );
+} );
+```
+**Note:** Be careful not to check in tests when using the `only()` feature, otherwise the build server will __only__ run those suites or tests.
+
+[eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest) runs over all tests files, and will catch common errors. Often your changes will affect other parts of the application, so it's a good idea to run all the unit tests locally before checking in.
+
+## Writing tests
+We use [Jest](https://facebook.github.io/jest) for testing our client-side application, and [Enzyme](https://github.com/airbnb/enzyme) for testing React components.
+
+Though you'll still see Chai assertions and Sinon spies/mocks throughout the code base, for all new tests we use the Jest API for [assertions](http://facebook.github.io/jest/docs/en/expect.html#content), [mocks](http://facebook.github.io/jest/docs/en/mock-functions.html#content) and [mock function/spies](http://facebook.github.io/jest/docs/en/mock-function-api.html#content).
```

Comment:
If we're specifically talking about both, should we mention that the jest expect function is still available at `global.expect` after `import { expect } from 'chai';`. Too much detail?

Reply:
Only if a suite requires two assertion libraries :), but I can't think of why aside from personal preference. 

We mention something similar (that we should use Jest from now on) on the [overview doc](https://github.com/Automattic/wp-calypso/blob/master/docs/testing/testing-overview.md). I thought the repetition was warranted, since ferreting through the code base for examples might cause some people to assume that both assertion libs are okay to use going forward.